lib/enumerable.rb

```diff
@@ -0,0 +1,95 @@
+module Enumerable
+  def my_each
+    idx = 0
+
+    arr ||= to_a
+    while idx < arr.length
+      yield(arr[idx])
+      idx += 1
+    end
+
+    arr
+  end
+
+  def my_each_with_index
+    idx = 0
+
+    while idx < length
+      yield(self[idx], idx)
+      idx += 1
+    end
+
+    self
+  end
```

Comment:
- [ ] Should return an Enumerator if no block is given 

- [ ]  when self is a range, it should return the range itself after calling the given block once for each element in self